### PR TITLE
[feature] add day01 husky

### PR DESCRIPTION
**Pull request by HuskyHsu**

### CODE REVIEW

File: day01/Husky/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    <KeyButton
+      key={key.letter}
+      name={key.name}
+      letter={key.letter}
+      playing={key.playing}
+      sound={key.sound}
+    />
+  ));
+
+  const keyPress = (type: string) => (event: { code: string }) => {
```
Comment (kayac-chang) [inline day01/Husky/src/App.tsx]: seems like this function is pure function, 
why not moving out from `Component`

File: day01/Husky/src/App.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+import soundList from "./assets/sound";
+import KeyButton from "./components/KeyButton";
+import { play } from "./utils/music";
+
+function App() {
+  const [data, setData] = useState(soundList);
+
+  const keys = soundList.map((key, index) => (
+    <KeyButton
```
Comment (kayac-chang) [inline day01/Husky/src/App.tsx]: move to line 38 improve readability

File: day01/Husky/src/assets/sound.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import boom from "../../../sample/sounds/boom.wav";
```
Comment (kayac-chang) [inline day01/Husky/src/assets/sound.ts]: if relative path become verbose, using path alias to improve

File: day01/Husky/src/components/KeyButton.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import Key from "../models/Key";
+
+export default function KeyButton(props: Key) {
+  return (
+    <div
+      className={
+        "border-4 border-solid border-black rounded-lg m-4 text-2xl py-4 px-2 w-40 " +
```
Comment (kayac-chang) [inline day01/Husky/src/components/KeyButton.tsx]: why not using `clsx`, or write a utility for compose class name

File: day01/Husky/src/models/Sound.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export default interface SoundMap {
```
Comment (kayac-chang) [inline day01/Husky/src/models/Sound.ts]: didn't found usage of this

File: day01/Husky/src/App.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    target.playing = type === "down";
+    setData([...data]);
+  };
+
+  const handleKeyPressDown = useCallback(keyPress("down"), data);
+  const handleKeyPressUp = useCallback(keyPress("up"), data);
+  useEffect(() => {
+    window.addEventListener("keydown", handleKeyPressDown);
+    window.addEventListener("keyup", handleKeyPressUp);
+  });
```
Comment (kayac-chang) [inline day01/Husky/src/App.tsx:32]: please add deps array, this will always run after component rerender

**Review by kayac-chang — APPROVED** [review]

File: day01/Husky/src/App.tsx (hunk 1 above)
Comment (HuskyHsu) [reply day01/Husky/src/App.tsx]: fix.
因為有用到useState的東西，我把他也改成function的參數了
感謝提醒！

File: day01/Husky/src/App.tsx (hunk 2 above)
Comment (HuskyHsu) [reply day01/Husky/src/App.tsx]: fixed

File: day01/Husky/src/assets/sound.ts (hunk 3 above)
Comment (HuskyHsu) [reply day01/Husky/src/assets/sound.ts]: fixed

File: day01/Husky/src/components/KeyButton.tsx (hunk 4 above)
Comment (HuskyHsu) [reply day01/Husky/src/components/KeyButton.tsx]: fixed

File: day01/Husky/src/models/Sound.ts (hunk 5 above)
Comment (HuskyHsu) [reply day01/Husky/src/models/Sound.ts]: removed

File: day01/Husky/src/App.tsx (hunk 6 above)
Comment (HuskyHsu) [reply day01/Husky/src/App.tsx:32]: fixed.
remove useCallback
add cleaning up an effect

**Pull request merged**